### PR TITLE
Build comprehensive tests

### PR DESCRIPTION
**Pull request by D-Limiter**

### CODE REVIEW

**Review by WangCheng0116 — APPROVED** [review]
LGTM

File: src/test/java/flashlingo/testutil/TypicalFlashCards.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+    public static final FlashCard WORD = new FlashCardBuilder().withOriginalWord("你好", "Chinese")
+            .withTranslatedWord("hi", "English").withLevel(2)
+            .withWhenToReview(Date.from(Instant.parse(DEFAULT_WHEN_TO_REVIEW))).build();
+
+    public static final String KEYWORD_MATCHING_MEIER = "Meier"; // A keyword that matches MEIER
+
+    private TypicalFlashCards() {} // prevents instantiation
+
+    /**
+     * Returns an {@code AddressBook} with all the typical persons.
```
Comment (itsNatTan) [inline src/test/java/flashlingo/testutil/TypicalFlashCards.java:112]: Please change this comment

File: src/test/java/flashlingo/logic/LogicManagerTest.java [hunk 2]
```diff
@@ -160,7 +162,7 @@ public void saveFlashlingo(ReadOnlyFlashlingo flashlingo, Path filePath)
                 new JsonUserPrefsStorage(temporaryFolder.resolve("ExceptionUserPrefs.json"));
         StorageManager storage = new StorageManager(flashlingoStorage, userPrefsStorage);
 
-        logic = new LogicManager(model, storage);
+        logic = (Logic) new LogicManager(model, storage);
 
         // Triggers the saveAddressBook method by executing an add command
```
Comment (itsNatTan) [inline src/test/java/flashlingo/logic/LogicManagerTest.java:167]: Change this comment

**Review by itsNatTan — CHANGES_REQUESTED** [review]
Minor nits, just change references to addressbook and person

**Pull request closed without merging**